src/components/WorldMap/WorldMap.jsx

```diff
@@ -0,0 +1,295 @@
+    this.projectPoint = this.projectPoint.bind(this);
+  }
+
+  componentDidMount() {
+    this.setup();
+  }
+
+  /*
+   * Convert d3 / geojson point to leaflet point
+   */
+  projectPoint(geo, x, y) {
+    // NEEDS the map here.
+    const point = this.map.latLngToLayerPoint(new L.LatLng(y, x)); // eslint-disable-line
+    geo.stream.point(point.x, point.y);
+  }
+
+  /**
+   * Setup Map, background layer, and other globals.
+   */
+  setup() {
+    const { location, zoom, updateFrequency } = this.props;
+    this.map = L.map(this.root, // eslint-disable-line
+        { maxZoom: 4, minZoom: 1 }
+    );
+
+    const layer = Tangram.leafletLayer({ // eslint-disable-line
+      scene: 'refill-style.yaml',
+      attribution: '<a href="https://mapzen.com/tangram" target="_blank">Tangram</a> | &copy; OSM contributors | <a href="https://mapzen.com/" target="_blank">Mapzen</a>',
+    });
+
+    this.map.setView(location, zoom);
+    layer.addTo(this.map);
+
+    this.svg = d3.select(this.map.getPanes().overlayPane).append('svg');
+    this.g = this.svg.append('g').attr('class', 'leaflet-zoom-hide');
+
+    // pointProject needs access to itself (this) and this's (that) map
+    const that = this;
+    const transform = d3.geoTransform({ point(...args) { that.projectPoint(this, ...args); } });
+
+    // Used in the update to rerender paths
+    this.path = d3.geoPath().projection(transform);
+
+    this.map.on('viewreset', this.updatePoints);
+
+    // update visual
+    this.timer = d3.interval(this.updateViewable, updateFrequency);
+  }
+
+
+  /**
+   * Update viewable index.
+   */
+  updateViewable() {
+    const { geoData } = this.props;
+
+    if (!geoData || geoData.features.length === 0) {
+      return;
+    }
+
+    this.viewableIndex += 1;
+
+    this.updatePoints();
+
+    if (this.viewableIndex >= geoData.features.length) {
+      this.timer.stop();
+    }
+  }
+
+  /**
+   * Redraw points on update or zoom
+   */
+  updatePoints() {
+    const { geoData, servers } = this.props;
+    // const { viewableIndex } = this.state;
+
+    if (!geoData || geoData.features.length === 0) {
+      return;
+    }
+
+    const bounds = this.path.bounds(geoData);
+    const topLeft = bounds[0];
+    const bottomRight = bounds[1];
+
+    this.svg.attr('width', bottomRight[0] - topLeft[0])
+      .attr('height', bottomRight[1] - topLeft[1])
+      .style('left', `${topLeft[0]}px`)
+      .style('top', `${topLeft[1]}px`);
+
+    this.g.attr('transform', `translate(${-topLeft[0]},${-topLeft[1]})`);
+
+    const pointScale = d3.scaleSqrt()
+      .domain([0, 100])
+      .range([2, 18])
+      .clamp(true);
+
+      // CLIENTS
+    this.path.pointRadius((d) => pointScale(d.properties.data.download_speed_mbps));
+
+    const viewable = geoData.features.slice(0, this.viewableIndex);
+
+    const points = this.g.selectAll('.client')
+      .data(viewable, (d) => d.id);
+
+    const pointsE = points.enter()
+      .append('path')
+      .classed('client', true)
```

Comment:
right - thats typically _why_ i do `classed` - so that i don't _have_ to think/worry about if there are other classes on an element (which typically there are not - so either way its no big deal).